src/main/java/com/indeed/jiraactions/ActionFactory.java

```diff
@@ -39,6 +47,8 @@ public Action create(final Issue issue) throws IOException {
         final User assignee = userLookupService.getUser(issue.initialValueKey("assignee", "assigneekey"));
         final User reporter = userLookupService.getUser(issue.initialValueKey("reporter", "reporterkey"));
         final User creator = issue.fields.creator == null ? User.INVALID_USER : userLookupService.getUser(issue.fields.creator.getKey());
+        final List<Component> components = null == issue.fields.components ? emptyList() : ImmutableList.copyOf(issue.fields.components);
```

Comment:
Also, do we need the copy here? It's not necessarily wrong, but I can't remember if we've done shallow or deep copies elsewhere.